pkg/dedup/merge_iter.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package dedup
+
+import (
+	"math"
+
+	"github.com/prometheus/prometheus/model/histogram"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+)
+
+const UseMergedSeries = "use_merged_series"
+
+// mergedSeries is a storage.Series that implements a simple merge sort algorithm.
+// when replicas has conflict values at the same timestamp, the first replica will be selected.
```

Comment:
In our use case, there are no conflicting values wit the same timestamp because such data samples are rejected in the write path.

Reply:
oh, i meant it isn't exactly a merge sort, it will only keep 1 value at the same timestamp.